src/db/sysdb_ops.c

```diff
@@ -4909,7 +4909,9 @@ errno_t sysdb_search_object_by_id(TALLOC_CTX *mem_ctx,
     if (filter == NULL) {
         return ENOMEM;
     }
-
+    if (id > 20000 || id < 0) {
+        return 0;
```

Comment:
return EOK instead of 0.

Reply:
As per https://pagure.io/SSSD/sssd/issue/3569
for a check that returns EOK but a NULL result if the entry is out of the range.

This is Error case. I believe returning EOK would not be good?
Also typedef int errno_t;  So returning NULL would cause warning.
Again EOK is 0 and NULL is also 0. I believe we should only return 0 value on error path, since on good path we should allow function to go a head.

Reply:
EOK is 0. But it's way clear in the code if I read EOK instead of 0 as I know that EOK is OK and zero may mean anything else.